.buildkite/scripts/steps/build_and_test.sh

```diff
@@ -0,0 +1,66 @@
+#!/bin/bash
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License
+# 2.0 and the following additional limitation. Functionality enabled by the
+# files subject to the Elastic License 2.0 may only be used in production when
+# invoked by an Elasticsearch process with a license key installed that permits
+# use of machine learning features. You may not use this file except in
+# compliance with the Elastic License 2.0 and the foregoing additional
+# limitation.
+#
+
+# This script gets run within the Docker container when a build is done in a
+# Docker container.
```

Comment:
Most of what's in this script looks very similar/identical to what's in `docker_entrypoint.sh`.

Could this file be changed to be a very thin wrapper around `docker_entrypoint.sh`?

Eventually BuildKite will have to be our CI system for _all_ supported branches, including 7.17, and 7.17 is not using CMake. So the BuildKite files will be easier to backport if they call other scripts that vary between branches rather than duplicating them.

If this is absolutely impossible then we can duplicate.  But another thing is that we'll need to maintain the ability to build locally in a Docker container for debugging and iterating on the Docker builds themselves.  If we have two scripts in two different directories that contain very similar code but one hardly ever gets used and the other gets used all the time then the one that hardly ever gets used will fall into disrepair.